test/e2e_node/oomkiller_test.go

```diff
@@ -32,32 +32,62 @@ import (
 )
 
 type testCase struct {
+	name                   string
 	podSpec                *v1.Pod
 	oomTargetContainerName string
 }
 
-const PodOOMKilledTimeout = 2 * time.Minute
-
 var _ = SIGDescribe("OOMKiller [LinuxOnly] [NodeConformance]", func() {
 	f := framework.NewDefaultFramework("oomkiller-test")
 	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
 
-	containerName := "oomkill-target-container"
-	oomPodSpec := getOOMTargetPod("oomkill-target-pod", containerName)
-	runOomKillerTest(f, testCase{podSpec: oomPodSpec, oomTargetContainerName: containerName})
+	testCases := []testCase{{
+		name:                   "single process container",
+		oomTargetContainerName: "oomkill-single-target-container",
+		podSpec: getOOMTargetPod("oomkill-target-pod", "oomkill-single-target-container",
+			getOOMTargetContainer),
+	}}
+
+	// If using cgroup v2, we set memory.oom.group=1 for the container cgroup so that any process which gets OOM killed
+	// in the process, causes all processes in the container to get OOM killed
+	if isCgroupV2() {
+		testCases = append(testCases, testCase{
+			name:                   "multi process container",
+			oomTargetContainerName: "oomkill-multi-target-container",
+			podSpec: getOOMTargetPod("oomkill-target-pod", "oomkill-multi-target-container",
+				getOOMTargetContainerMultiProcess),
+		})
+	}
+	for _, tc := range testCases {
+		runOomKillerTest(f, tc)
+	}
 })
 
+// isCgroupV2 is a hack to check for cgroup v2 without importing a platform specific library
+func isCgroupV2() bool {
```

Comment:
a separate nit here is there's https://github.com/opencontainers/runc/blob/v1.1.7/libcontainer/cgroups/utils.go#L34 which achieves this more precisely. Why not use that?

Reply:
IIRC, I tried that first but the test fails to build on Windows.  I checked and all usage of libcontainercgroups are in `*_linux.go`

Reply:
Yeah, I confirmed it fails that way:

```
$ GOOS=windows go test -v
# github.com/opencontainers/runc/libcontainer/devices
../../vendor/github.com/opencontainers/runc/libcontainer/devices/device.go:173:9: undefined: mkDev
```

Reply:
I feel like this file should be linux only.. the test says [LinuxOnly]...

Reply:
doesn't have to be addressed here but could be if you felt so inclined :)

Reply:
Easy enough and lets me remove some code.